addons/mrp/models/mrp_workorder.py

```diff
@@ -650,20 +650,24 @@ def button_start(self, raise_on_invalid_state=False):
     def button_finish(self):
         date_finished = fields.Datetime.now()
         all_vals_dict = defaultdict(lambda: self.env['mrp.workorder'])
+        workorders_to_end = self.env['mrp.workorder']
+        moves = self.move_raw_ids + self.production_id.move_byproduct_ids.filtered(lambda m: (m.operation_id == self.operation_id))
```

Comment:
on multiple byproducts linked to different workorders, none will be selected or picked
except if called from `button_mark_done`, thanks to `_post_inventory`